demux/stheader.h

```diff
@@ -41,6 +41,7 @@ struct sh_stream {
 
     struct mp_codec_params *codec;
 
+    int program_num;
```

Comment:
Calling av_find_program_from_stream is slow and annoying. I also need to fetch the program when the stream first shows up and store it, because later the stream might not be in the program anymore after the PMT updates.